plugins/hls-change-type-signature-plugin/src/Ide/Plugin/ChangeTypeSignature.hs

```diff
@@ -67,39 +111,117 @@ data ChangeSignature = ChangeSignature {
+        mismatch <- solverReport ^? _MismatchMessage
+        expectedType <- mismatch ^? _TypeEqMismatchExpected
+        actualType <- mismatch ^? _TypeEqMismatchActual
+
+        pure (showType expectedType, showType actualType, errInfo)
 
+    logWith recorder Debug (LogErrInfoCtxt errInfo)
+
+    -- Extract the declName from the extra error text
+    declName <- hoistMaybe (matchingDiagnostic errInfo)
+
+    -- Look up location of declName. If it fails, log it
+    declSrcSpan <-
+        case findSigLocOfStringDecl decls expectedType (T.unpack declName) of
+            Just x -> pure x
+            Nothing -> do
+                logWith recorder Debug (LogFindSigLocFailure declName)
+                hoistMaybe Nothing
+
+    pure ChangeSignature{..}
+    where
+        showType :: Type -> Text
+        showType = T.pack . showSDocUnsafe . pprTidiedType
+
+_TcRnMessageDetailed :: Traversal' TcRnMessage TcRnMessageDetailed
+_TcRnMessageDetailed focus (TcRnMessageWithInfo errInfo detailed) =
+    (\detailed' -> TcRnMessageWithInfo errInfo detailed') <$> focus detailed
+_TcRnMessageDetailed _ msg = pure msg
+
+_TcRnSolverReport :: Traversal' TcRnMessage SolverReportWithCtxt
+#if MIN_VERSION_ghc(9,10,0)
+_TcRnSolverReport focus (TcRnSolverReport report reason) =
+    (\report' -> TcRnSolverReport report' reason) <$> focus report
+#else
+_TcRnSolverReport focus (TcRnSolverReport report reason hints) =
+    (\report' -> TcRnSolverReport report' reason hints) <$> focus report
+#endif
+_TcRnSolverReport _ msg = pure msg
+
+tcSolverReportMsgL :: Lens' SolverReportWithCtxt TcSolverReportMsg
+tcSolverReportMsgL = lens reportContent (\report content' -> report { reportContent = content' })
+
+_MismatchMessage :: Traversal' TcSolverReportMsg MismatchMsg
+_MismatchMessage focus (Mismatch msg t a c) = (\msg' -> Mismatch msg' t a c) <$> focus msg
+_MismatchMessage focus (CannotUnifyVariable msg a) = flip CannotUnifyVariable a <$> focus msg
+_MismatchMessage _ report = pure report
+
+_TypeEqMismatchExpected :: Traversal' MismatchMsg Type
+#if MIN_VERSION_ghc(9,12,0)
+_TypeEqMismatchExpected focus mismatch@(TypeEqMismatch _ _ _ expected _ _ _) =
+    (\expected' -> mismatch { teq_mismatch_expected = expected' }) <$> focus expected
+#else
+_TypeEqMismatchExpected focus mismatch@(TypeEqMismatch _ _ _ _ expected _ _ _) =
+    (\expected' -> mismatch { teq_mismatch_expected = expected' }) <$> focus expected
+#endif
+_TypeEqMismatchExpected _ mismatch = pure mismatch
+
+_TypeEqMismatchActual :: Traversal' MismatchMsg Type
+#if MIN_VERSION_ghc(9,12,0)
+_TypeEqMismatchActual focus mismatch@(TypeEqMismatch _ _ _ _ actual _ _) =
+    (\actual' -> mismatch { teq_mismatch_actual = actual' }) <$> focus actual
+#else
+_TypeEqMismatchActual focus mismatch@(TypeEqMismatch _ _ _ _ _ actual _ _) =
+    (\actual' -> mismatch { teq_mismatch_expected = actual' }) <$> focus actual
+#endif
+_TypeEqMismatchActual _ mismatch = pure mismatch
```

Comment:
I think these should live in ` Development.IDE.GHC.Compat.Error`, what do you think about that?